x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/grid_embeddable/grid_embeddable.tsx

```diff
@@ -0,0 +1,758 @@
+  }
+
+  useEffect(() => {
+    loadOverallStats();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchQuery, samplerShardSize, lastRefresh]);
+
+  useEffect(() => {
+    createMetricCards();
+    createNonMetricCards();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [overallStats, showEmptyFields]);
+
+  useEffect(() => {
+    loadMetricFieldStats();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [metricConfigs]);
+
+  useEffect(() => {
+    loadNonMetricFieldStats();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [nonMetricConfigs]);
+
+  useEffect(() => {
+    createMetricCards();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [metricsLoaded]);
+
+  useEffect(() => {
+    createNonMetricCards();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [nonMetricsLoaded]);
+
+  const configs = useMemo(() => {
+    let combinedConfigs = [...nonMetricConfigs, ...metricConfigs];
+    if (visibleFieldTypes && visibleFieldTypes.length > 0) {
+      combinedConfigs = combinedConfigs.filter(
+        (config) => visibleFieldTypes.findIndex((field) => field === config.type) > -1
+      );
+    }
+    if (visibleFieldNames && visibleFieldNames.length > 0) {
+      combinedConfigs = combinedConfigs.filter(
+        (config) => visibleFieldNames.findIndex((field) => field === config.fieldName) > -1
+      );
+    }
+
+    return combinedConfigs;
+  }, [nonMetricConfigs, metricConfigs, visibleFieldTypes, visibleFieldNames]);
+
+  // Some actions open up fly-out or popup
+  // This variable is used to keep track of them and clean up when unmounting
+  const actionFlyoutRef = useRef<() => void | undefined>();
+  useEffect(() => {
+    const ref = actionFlyoutRef;
+    return () => {
+      // Clean up any of the flyout/editor opened from the actions
+      if (ref.current) {
+        ref.current();
+      }
+    };
+  }, []);
+
+  // Inject custom action column for the index based visualizer
+  // Hide the column completely if no access to any of the plugins
+  const extendedColumns = useMemo(() => {
+    const actions = getActions(
+      input.indexPattern,
+      { lens: services.lens },
+      {
+        searchQueryLanguage,
+        searchString,
+      },
+      actionFlyoutRef
+    );
+    if (!Array.isArray(actions) || actions.length < 1) return;
+
+    const actionColumn: EuiTableActionsColumnType<FieldVisConfig> = {
+      name: (
+        <FormattedMessage
+          id="xpack.dataVisualizer.index.dataGrid.actionsColumnLabel"
+          defaultMessage="Actions"
+        />
+      ),
+      actions,
+      width: '70px',
+    };
+
+    return [actionColumn];
+  }, [input.indexPattern, services, searchQueryLanguage, searchString]);
+
+  return {
+    configs,
+    searchQueryLanguage,
+    searchString,
+    searchQuery,
+    extendedColumns,
+    documentCountStats,
+    metricsStats,
+  };
+};
+
+export const DiscoverWrapper = ({
```

Comment:
QQ: ist this a special wrapper for Discover? then it might make sense to move this code to the Discover plugin ... in other case I think it should be labeled in a way that other plugin/apps are not afraid to use it :)

Reply:
Updated here https://github.com/elastic/kibana/pull/107184/commits/ba1cad7ff6ee7b3391fb42000306c6b3b96096d5

Reply:
thx!